BrowserKit/Sources/TabDataStore/TabSessionStore.swift

```diff
@@ -10,29 +10,30 @@ public protocol TabSessionStore {
     /// - Parameters:
     ///   - tabID: an ID that uniquely identifies the tab
     ///   - sessionData: the data associated with a session, encoded as a Data object
-    func saveTabSession(tabID: UUID, sessionData: Data) async
+    func saveTabSession(tabID: UUID, sessionData: Data)
 
     /// Fetches the session data associated with a tab
     /// - Parameter tabID: an ID that uniquely identifies the tab
     /// - Returns: the data associated with a session, encoded as a Data object
-    func fetchTabSession(tabID: UUID) async -> Data?
+    func fetchTabSession(tabID: UUID) -> Data?
 
     /// Cleans up any tab session data files for tabs that are no longer open.
     func deleteUnusedTabSessionData(keeping: [UUID]) async
 }
 
-public actor DefaultTabSessionStore: TabSessionStore {
+public class DefaultTabSessionStore: TabSessionStore {
     let fileManager: TabFileManager
     let logger: Logger
     let filePrefix = "tab-"
+    private let lock = NSRecursiveLock()
 
     public init(fileManager: TabFileManager = DefaultTabFileManager(),
                 logger: Logger = DefaultLogger.shared) {
         self.fileManager = fileManager
         self.logger = logger
     }
 
-    public func saveTabSession(tabID: UUID, sessionData: Data) async {
+    public func saveTabSession(tabID: UUID, sessionData: Data) {
```

Comment:
Because of this, `saveSessionData` method no longer needs a Task await on line 312 of `TabManagerImplementation.swift`.

Reply:
Thank you, fixed